specification/azurearcdata/resource-manager/Microsoft.AzureArcData/preview/2025-06-01-preview/sqlServerAvailabilityGroups.json

```diff
@@ -0,0 +1,1932 @@
+          "readOnly": true
+        },
+        "suspendReasonDescription": {
+          "type": "string",
+          "description": "Description of the database suspended state reason.",
+          "readOnly": true
+        }
+      }
+    },
+    "SqlAvailabilityGroupStaticIPListenerProperties": {
+      "type": "object",
+      "description": "The properties of a static IP Arc Sql availability group listener",
+      "properties": {
+        "dnsName": {
+          "type": "string",
+          "description": "the DNS name for the listener."
+        },
+        "ipV4AddressesAndMasks": {
+          "$ref": "#/definitions/SqlAvailabilityGroupIpV4AddressesAndMasksProperties",
+          "description": "IP V4 Addresses and masks for the listener."
+        },
+        "ipV6Addresses": {
+          "type": "array",
+          "description": "IP V6 Addresses for the listener",
+          "items": {
+            "type": "string"
+          }
+        },
+        "port": {
+          "type": "integer",
+          "description": "Network port for the listener. Default is 1433.",
+          "format": "int32"
+        }
+      }
+    },
+    "SqlAvailabilityGroupIpV4AddressesAndMasksProperties": {
+      "type": "array",
+      "description": "Address and netmask information for an IPv4 AG listener.",
+      "items": {
+        "type": "object",
+        "properties": {
+          "ipAddress": {
+            "type": "string",
+            "description": "IPV4 address"
+          },
+          "mask": {
+            "type": "string",
+            "description": "IPV4 netmask"
+          }
+        }
+      },
+      "x-ms-identifiers": [
+        "ipAddress"
+      ]
+    },
+    "SqlAvailabilityGroupDhcpListenerProperties": {
+      "type": "object",
+      "description": "The properties of a DHCP Arc Sql availability group listener",
+      "properties": {
+        "dnsName": {
+          "type": "string",
+          "description": "the DNS name for the listener."
+        },
+        "subnet": {
+          "type": "string",
+          "description": "The IPV4 subnet for the listener."
+        },
+        "mask": {
+          "type": "string",
+          "description": "The netmask for the listener."
+        },
+        "port": {
+          "type": "integer",
+          "description": "Network port for the listener. Default is 1433.",
+          "format": "int32"
+        }
+      }
+    },
+    "FailoverMiLinkResourceId": {
+      "description": "The azure resource identifier for the Sql Managed Instance.",
+      "type": "object",
+      "properties": {
+        "managedInstanceId": {
+          "type": "string",
+          "description": "Azure resource id for the sql managed instance.",
+          "format": "arm-id"
+        },
+        "force": {
+          "type": "boolean",
+          "description": "Whether to perform a Force Failover as opposed to just a regular Failover."
+        }
```

Comment:
Enums are highly recommended over booleans. 

Please model this as an enum to allow for multiple values and future proofing.